src/vset.c

```diff
@@ -0,0 +1,2094 @@
+
+    for (uint32_t i = 0; i < pv->len; i++) {
+        if (pv->data[i] == elem) {
+            *ppv = pvRemoveAt(pv, i);
+            return true;
+        }
+    }
+    return false;
+}
+
+/* Retrieves the element at the specified index in the pVector.
+ *
+ * Arguments:
+ *   vec - The pVector to retrieve from.
+ *   idx - The index of the element to access.
+ *
+ * Return:
+ *   A pointer to the element at the given index.
+ *   Returns NULL if the vector is NULL or the index is out of bounds. */
+void *pvGet(pVector *vec, uint32_t idx) {
+    assert(vec && idx < vec->len);
+    return vec->data[idx];
+}
+
+/* Frees the memory used by the pVector.
+ *
+ * Arguments:
+ *   pv - The pVector to free.
+ *
+ * Return:
+ *   None. */
+void pvFree(pVector *pv) {
+    if (pv) zfree(pv);
+}
+
+/* Appends an element to the end of the given pVector.
+ *
+ * Parameters:
+ *   pv   - The vector to append to.
+ *   elem - The element to append.
+ *
+ * Returns:
+ *   A (possibly reallocated) pVector with the new element inserted at the end.
+ *
+ * Notes:
+ *   Internally this uses pvInsert() with the current length of the vector,
+ *   effectively appending the element. */
+pVector *pvPush(pVector *pv, void *elem) {
+    return pvInsert(pv, elem, PV_LEN(pv));
+}
+
+/* Removes and optionally returns the last element from the given pVector.
+ *
+ * Parameters:
+ *   pv    - The vector to remove the element from.
+ *   pelem - Optional pointer to store the popped element. Can be NULL.
+ *
+ * Returns:
+ *   A (possibly reallocated) pVector with the last element removed.
+ *
+ * Notes:
+ *   If the vector is empty, the behavior is to remove from index 0 (safe fallback).
+ *   You can pass NULL for `pelem` if you don't need the removed value. */
+pVector *pvPop(pVector *pv, void **pelem) {
+    uint32_t last_idx = PV_LEN(pv) > 0 ? PV_LEN(pv) - 1 : 0;
+    if (pelem) *pelem = pvGet(pv, last_idx);
+    return pvRemoveAt(pv, last_idx);
+}
+
+/* Set the element at given index inside the pVector.
+ *
+ * Parameters:
+ *   pv    - The vector containing the elements to swap.
+ *   idx   - Index of the element.
+ *   elem  - pointer to the new element.
+ *
+ * Returns:
+ *   None.
+ *
+ * Preconditions:
+ *   - idx must be valid indices within the vector. */
+void pvSet(pVector *pv, uint32_t idx, void *elem) {
+    assert(idx < PV_LEN(pv));
+    pv->data[idx] = elem;
+}
+
+/* Swaps two elements at given indices inside the pVector.
+ *
+ * Parameters:
+ *   pv    - The vector containing the elements to swap.
+ *   idx1  - Index of the first element.
+ *   idx2  - Index of the second element.
+ *
+ * Returns:
+ *   None.
+ *
+ * Preconditions:
+ *   - idx1 and idx2 must both be valid indices within the vector.
+ *
+ * Notes:
+ *   This is a simple in-place swap that uses direct pointer assignment. */
+void pvSwap(pVector *pv, uint32_t idx1, uint32_t idx2) {
+    assert(idx1 < PV_LEN(pv) && idx2 < PV_LEN(pv));
+    void *temp = pvGet(pv, idx1);
```

Comment:
Odd to use `pvGet` for this line, but not use `pvGet` or `pvSet` for other lines.
```suggestion
    void *temp = pv->data[idx1];
```